kiwiproject/kiwi-test
# Add MongoDbExtension

* Add MongoDbExtension JUnit Jupiter extension
* Rename HostDomainBehavior enum in  MongoTestProperties
  to ServiceHostDomain to make more clear that it relates to
  the service host (and not the MongoDB host)
* Update/refactor MongoTestProperties constructor to properly pass
  the non-null serviceHostDomainBehavior to serviceHost, and also to
  pass the normalized service host to unitTestDatabaseName
* Add null arg check to MongoTestProperties#serviceHost
* Add instance method getDatabaseNameWithoutTimestamp() in
  MongoTestProperties which delegates to the static utility
  databaseNameWithoutTimestamp method
* Add instance method getDatabaseTimestamp() in MongoTestProperties
  which delegates to the static utility extractDatabaseTimestamp method
* Add test dependency for in-memory Mongo server for use by tests
* Update the docs in MongoTestProperties to reference MongoDbExtension
* Add multiple integration tests for MongoDbExtension which test the
  behavior when the test databases are dropped `@AfterAll` (default),
  `@BeforeEach`, and `@AfterEach`, as well as testing the `NEVER`
  collection cleanup option.
* Add unit test of MongoDbExtension that tests construction using the
  constructors and the Lombok builder, as well as cleanup of databases
  from previous test executions
* Add MongoDbTestHelpers as a helper class for the integration tests

Fixes #123
